server/src/main/java/io/crate/planner/operators/LogicalPlanner.java

```diff
@@ -336,14 +343,29 @@ public LogicalPlan visitUnionSelect(UnionSelect union, List<Symbol> outputs) {
 
         @Override
         public LogicalPlan visitQueriedSelectRelation(QueriedSelectRelation relation, List<Symbol> outputs) {
+            Predicate<Symbol> isCorrelatedSubquery =
+                s -> s instanceof SelectSymbol selectSymbol && selectSymbol.isCorrelated();
+            if (SymbolVisitors.any(isCorrelatedSubquery, relation.where())) {
+                throw new UnsupportedOperationException(
+                    "Using correlated subqueries in the WHERE clause is not supported");
+            }
             SplitPoints splitPoints = SplitPointsBuilder.create(relation);
             LogicalPlan source = JoinPlanBuilder.buildJoinTree(
                 relation.from(),
                 relation.where(),
                 relation.joinPairs(),
                 rel -> {
                     if (relation.from().size() == 1) {
-                        return rel.accept(this, splitPoints.toCollect());
+                        // TODO: Move this elsewhere?
+                        // Or resolve ScopedSymbol to SimpleReference eagerly in the analyzer,
+                        // given that OuterColumn gives the context?
+                        ArrayList<Symbol> toCollect = new ArrayList<>(splitPoints.toCollect());
+                        for (var outerColumn : splitPoints.outerColumns()) {
+                            var columnIdent = Symbols.pathFromSymbol(outerColumn.symbol());
+                            Symbol field = rel.getField(columnIdent, Operation.READ);
+                            toCollect.add(field);
+                        }
```

Comment:
Why do we need to re-map the `outerColumn.symbol()` here? Isn't it already a  (Scoped)Symbol retrieved over the parent's relation at the `FullQualifiedNameFieldProvider` (https://github.com/crate/crate/pull/12851/files#diff-8c424f41db36fb37078c2cd7e6d0dbc3149720c11390d977d8a1e6ab242d9062R150)?

Reply:
I think I had an error in my reasoning. I'll double check.

Reply:
Right it's no longer needed. I made the SplitPointBuilder changes later. Thanks for pointing out.